scripts/check-component-features

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -o errexit
+
+# Extract all feature names from Cargo.toml
+features=$(
+  sed -e '
+    1,/^\[features\]$/d;
+    /^\[/,$d;
+    /=/!d;
+    s/ *=.*$//;
+    /-utils-/d;
+    /^default$/d;
+    /^all-integration-tests$/d;
+  ' < Cargo.toml | sort
+)
+
+procs=$( nproc )
+
+# Prime the pump to build most of the artifacts
+cargo check --workspace --all-targets --no-default-features
+cargo check --workspace --all-targets --no-default-features --features default
+cargo check --workspace --all-targets --no-default-features --features all-integration-tests
+
+# The feature builds already run in parallel below, don't overload
+export CARGO_BUILD_JOBS=1
+
+exec xargs --max-procs=$procs --max-lines=1 scripts/check-one-feature <<< $features
```

Comment:
Just a note that I think one advantage of `parallel` is that it handles output interleaving for you. I think this is fine as-is though.